### DISCUSSION TITLE
File upload problems when combining livewire 3 with tenancy

### DISCUSSION THREAD

**Discussion by dweerd** [livewire/livewire#6878] [7. Bugs (archived)] [answered]
### Livewire version

3.0

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

1. Setup Laravel with Tenancy (https://github.com/archtechx/tenancy) and Livewire 3.0. I use a subdomain per tenant.
2. On a tenant-based route, create a component and use the trait `WithFileUploads`
3. Add a file upload field like `<input type="file" wire:model="image">`
4. Select a file

By default, livewire is configured to use `throttle:60,1` for the file upload url (`temporary_file_upload.middleware` configuration). But this throws an exception saying it can't find the `users` table. This is because there is no middleware configured which initializes tenancy, and my central (non-tenant) application does indeed not have a `users` table.

By changing the configuration to use a middleware group which includes both the the tenancy initialization middleware and the throttle middleware, the file is actually uploaded. But now the next problem is, that the preview doesn't work. The `/livewire/preview-file` endpoint throws a 500 error. This makes sense, as this endpoint also doesn't initialize the tenancy, and ends up looking for the file in the wrong folder.

When I update the route definition in `Livewire\Features\SupportFileUploads` to
```
Route::get('/livewire/preview-file/{filename}', [FilePreviewController::class, 'handle'])
            ->name('livewire.preview-file')
            ->middleware('web', \Stancl\Tenancy\Middleware\InitializeTenancyByDomain::class);
```
it actually works... But I guess this cannot be configured somehow?

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment, marked as answer]
@dweerd thanks for reporting! There's an open PR for being able to customise the file upload route, see PR #6803

**Reply by dweerd** [reply]
thanks for pointing that out!
